src/node_url.cc

```diff
@@ -255,21 +250,73 @@ void UpdateUrl(const FunctionCallbackInfo<Value>& args) {
       undef,
       undef,
       undef,
-      undef,
-      undef,
   };
   SetArgs(env, argv, out);
   USE(success_callback_->Call(
       env->context(), args.This(), arraysize(argv), argv));
   args.GetReturnValue().Set(result);
 }
 
+void FormatUrl(const FunctionCallbackInfo<Value>& args) {
+  CHECK_GT(args.Length(), 4);
+  CHECK(args[0]->IsString());  // url href
+
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+
+  Utf8Value href(isolate, args[0].As<String>());
+  const bool fragment = args[1]->IsTrue();
+  const bool unicode = args[2]->IsTrue();
+  const bool search = args[3]->IsTrue();
+  const bool auth = args[4]->IsTrue();
+
+  ada::result out = ada::parse(std::string_view(href.out(), href.length()));
```

Comment:
Just to say, when https://github.com/nodejs/node/pull/46723 lands, this can be simplified to:

```suggestion
  ada::result out = ada::parse(href.ToStringView());
```

Reply:
You're right. Nice!

Reply:
I rebased and updated my implementation.